### ISSUE TITLE
TypeError: QgsRubberBand(): argument 2 has unexpected type

### ISSUE THREAD

**Issue by hillaxj** [jawensi/DEMto3D-QGIS-Plugin#40]
When selecting print extent, "TypeError: QgsRubberBand(): argument 2 has unexpected type" appears. QgsRubberBand() has been changed to no longer accept bool or int vars for the 2nd argument. https://qgis.org/pyqgis/3.4/gui/QgsRubberBand.html 

 ```
An error has occurred while executing Python code: 

TypeError: QgsRubberBand(): argument 2 has unexpected type 'bool' 
Traceback (most recent call last):
  File "C:\Users/User/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D\DEMto3D_Dialog\DEMto3D_dialog.py", line 392, in full_extent
    self.paint_extent(self.rect_Params)
  File "C:\Users/User/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D\DEMto3D_Dialog\DEMto3D_dialog.py", line 552, in paint_extent
    self.extent = QgsRubberBand(self.canvas, True)
TypeError: QgsRubberBand(): argument 2 has unexpected type 'bool'

```
Python version: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]

**Comment by harmony365** [comment]
Cuando ejecuto la opción de selección me genera este error:

Ha ocurrido un error mientras se ejecutaba el código de Python: 

TypeError: QgsRubberBand(): argument 2 has unexpected type 'int' 
Traceback (most recent call last):
  File "C:\Users/MSI/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D-QGIS-Plugin-master\DEMto3D_Dialog\DEMto3D_dialog.py", line 447, in custom_extent
    self.rect_map_tool = RectangleMapTool(
  File "C:\Users/MSI/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D-QGIS-Plugin-master\DEMto3D_Dialog\DEMto3D_dialog.py", line 913, in __init__
    self.rubberBand = QgsRubberBand(self.canvas, 3)
TypeError: QgsRubberBand(): argument 2 has unexpected type 'int'


Versión de Python: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] 
Versión de QGIS: 3.30.1-'s-Hertogenbosch 's-Hertogenbosch, 9035a01e

**Comment by franpersanchez** [comment]
I experienced the same error described above.
I made a pull request with my proposal changes according to QGIS docs. The selection feature works fine now, generates the polygon and correctly shows the coordenates values.

**Comment by Oop-o** [comment]
Hey Sanchez, thank you for contributing... I downloaded your changes to the dialog.py file but I am still getting the same error message. Interestingly enough the error message still says 

_TypeError: QgsRubberBand(): argument 2 has unexpected type 'bool' 
Traceback (most recent call last):
  File "C:\Users/18174/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D\DEMto3D_Dialog\DEMto3D_dialog.py", line 392, in full_extent
    self.paint_extent(self.rect_Params)
  File "C:\Users/18174/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\DEMto3D\DEMto3D_Dialog\DEMto3D_dialog.py", line 552, in paint_extent
    self.extent = QgsRubberBand(self.canvas, True)
TypeError: QgsRubberBand(): argument 2 has unexpected type 'bool'_

Which conflicts with the code I actually have in my .py file at line 552 which is 

self.roi_x_min, self.roi_y_min),

I have checked all instances of QgsRubberBand() but they are all set to "QgsRubberBand(self.canvas, Qgis.GeometryType.Polygon)". I am not sure what is going on if you could advise that would be really appreaciated.

**Comment by Oop-o** [comment]
It looks like selecting print extent works actually... the error message is appearing for no reason.

**Comment by jawensi** [comment]
Fixed in new version 3.6
